operator/internal/controller/transaction_reconcile.go

```diff
@@ -215,9 +230,40 @@ func getPaladinRPC(ctx context.Context, c client.Client, nodeName, namespace str
 	if err != nil {
 		return nil, err
 	}
-	return pldclient.New().HTTP(ctx, &pldconf.HTTPClientConfig{
+
+	// Adding the timeout to the cache key to ensure that different timeouts are cached separately
+	// This is important because the timeout is used in the HTTP client config
+	// and different timeouts may require different configurations.
+	key := fmt.Sprintf("%s-%s", url, timeout)
+
+	// Check if the client is already in the cache
+	// Use a read lock to avoid blocking other goroutines
+	cache.mux.RLock()
+	if client, ok := cache.clients[key]; ok && client != nil {
+		cache.mux.RUnlock()
+		return client, nil
+	}
+	cache.mux.RUnlock()
+
+	// If not, create a new client and store it in the cache
+	cache.mux.Lock()
+	defer cache.mux.Unlock()
+
+	// Check again in the cache after acquiring the lock
+	// This is to ensure that another goroutine didn't create the client while we were waiting for the lock
+	if client, ok := cache.clients[key]; ok && client != nil {
+		return client, nil
+	}
+
+	client, err := pldclient.New().HTTP(ctx, &pldconf.HTTPClientConfig{
 		URL:               url,
 		ConnectionTimeout: confutil.P(timeout),
 		RequestTimeout:    confutil.P(timeout),
 	})
+	if err != nil {
+		return nil, err
+	}
+
+	cache.clients[key] = client
```

Comment:
Do we ever know when we could let go / release a client ? I.e. on `Paladin` delete do we need to check if the manager has a client for that node and can remove it as a result ?

Reply:
This is a good point, I will need to think about it.
Also, I need to check what happens to the client in case the paladin pod restarts. If the client can reconnect easily, so the current implementation is fine because the projected number of clients is low enough that there is no real need in a "cleanup" mechanism. However, if the client has trouble reconnecting after a pod restart than we have a bigger problem to fix.

Reply:
I added a deletion mechanism, although I'm not sure it's necessary given the small number of clients.